storage/replica.go

```diff
@@ -3232,6 +3238,7 @@ func (r *Replica) executeBatch(
 				} else {
 					ba.Timestamp.Forward(tErr.ActualTimestamp)
 				}
+				r.store.Clock().Update(tErr.ActualTimestamp)
```

Comment:
That's a good point. It was added in response to a comment from @tschottdorf:

> There's an awkward maybe-caveat here: if we update in a loop with each request, we pick up any forwarding that was made from the response of the previously executed one (or not? perhaps not), and we lose that with the code you remove below. I don't think we've ever thought about this particularly hard, but we should now that we're trimming the fat. Can MVCC forward timestamps in any way that puts them ahead of the clock? I suppose it will on WriteTooOld, where it adds a logical tick.